reconcile/gabi_authorized_users.py

```diff
@@ -0,0 +1,96 @@
+import sys
+from datetime import datetime, date
+from typing import Iterable, Mapping, Optional
+
+from reconcile.utils.aggregated_list import RunnerException
+import reconcile.queries as queries
+import reconcile.openshift_base as ob
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.openshift_resource import (OpenshiftResource as OR,
+                                                ResourceInventory as RI)
+from reconcile.utils.defer import defer
+
+QONTRACT_INTEGRATION = 'gabi-authorized-users'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+EXPIRATION_MAX = 90
+
+
+def construct_gabi_oc_resource(name: str, users: Iterable[str]) -> OR:
+    body = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {
+            "name": name,
+            "annotations": {
+                "qontract.recycle": "true"
+            }
+        },
+        "data": {
+            "authorized-users.yaml": '\n'.join([u for u in users])
+        }
+    }
+    return OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION,
+              error_details=name)
+
+
+def fetch_desired_state(gabi_instances: Iterable[Mapping], ri: RI) -> None:
+    for g in gabi_instances:
+        exp_date = datetime.strptime(g['expirationDate'], '%Y-%m-%d').date()
+        if exp_date < date.today():
+            continue
+        elif (exp_date - date.today()).days > EXPIRATION_MAX:
+            raise RunnerException(
+                f'The maximum expiration date of {g["name"]} '
+                f'shall not exceed {EXPIRATION_MAX} days form today')
+        users = [u['github_username'] for u in g['users']]
+        resource = construct_gabi_oc_resource(g['name'], users)
+        for i in g['instances']:
+            namespace = i['namespace']
+            account = i['account']
+            identifier = i['identifier']
+            tf_resources = namespace['terraformResources']
+            found = False
+            for tf_resource in tf_resources:
+                if tf_resource['provider'] == 'rds' and \
+                        tf_resource['account'] == account and \
+                        tf_resource['identifier'] == identifier:
+                    found = True
+                    break
+            if not found:
+                raise RunnerException(
+                    f'Could not find rds identifier {identifier} '
+                    f'for account {account} in namespace {namespace["name"]}')
+            cluster = namespace['cluster']['name']
+            ri.add_desired(
```

Comment:
Please remember that `add_desired` can raise a `KeyError`, if you really don't care, please suppress it.